### PR TITLE
Cockpit 277

### PR DESCRIPTION
**Pull request by garrett**
Preview: https://garrett.github.io/cockpit-project.github.io/blog/cockpit-277.html

TODO:
- [x] Add improvement password management blurb (it will be included in via a tag)
- [x] Regen in non-preview mode
- [x] Re-update bodhi links after regeneration. Links can be found @ https://raw.githubusercontent.com/cockpit-project/cockpit-project.github.io/328e76d04b87793d7d5d95c47b4ea48416134a82/_posts/2022-09-21-cockpit-277.md
- [x] Revise English text
- [x] Work on screenshots & video

### CODE REVIEW

**Comment by martinpitt** [comment]
It'd be good to mention the "browser is too old" fix from https://github.com/cockpit-project/cockpit/pull/17726 -- I added a release-note label there.

**Comment by garrett** [comment]
@martinpitt: Good idea. Wihle it's a "we fixed a bug", it could theoretically impact a lot of people if browsers are updated and their Cockpit install is not.

**Comment by skobyda** [comment]
Added release notes to the https://github.com/cockpit-project/cockpit/pull/17726 . Feel free to reword though.

**Comment by skobyda** [comment]
Updated bodhi links, regenerated notes

File: _posts/2022-09-22-cockpit-277.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+_Thanks to [Jacek Tomasiak](https://github.com/skazi0) for pointing this out and providing a fix!_
+
+## login: Adjust selectors when checking for `:is()` and `:where()` support
+
+Newer, upcoming browser versions have improved next-level support for `:is()` and `:where()` selectors. Cockpit was checking for support an empty usage which passed on browsers in the earlier (current as of writing) implementation. However, browsers have recently updated their parsing support for "[Forgiving Selector Parsing](https://developer.mozilla.org/en-US/docs/Web/CSS/:is#forgiving_selector_parsing)", which caused the newer development versions of Firefox, Chrome, and WebKit to fail this check, preventing the browsers from logging into Cockpit.
+
+The check has now been adjusted so current and upcoming browser versions all pass.
+
+Additionally, hotfixes for older supported affected versions of Cockpit have been published for CentOS, Debian, and Ubuntu. If you have an error while trying to log in with a new browser on an older Cockpit version, please upgrade your version of Cockpit.
```
Comment (martinpitt) [inline _posts/2022-09-22-cockpit-277.md:29]: This is a bit more complicated:

  - CentOS 8/9 stream are fixed.
  - RHEL 8.6/9.0 are affected, fixes not yet published. Older RHEL versions are *not* affected.
  - Debian got fixed
  - Ubuntu Kinetic (current devel series) and the backport for 22.04 LTS got fixed. The released version in 22.04 LTS is still affected, and will be fixed (I didn't start the process yet).

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks!

File: _posts/2022-09-22-cockpit-277.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
+We release regularly.
+
+Here are the release notes from Cockpit 277, cockpit-machines 275, and cockpit-podman 54:
+
+
+## login: Improve password manager compatibility
+
+The HTML of the login page has been adjusted to be more compatible with password managers in popular browsers. Usernames and passwords are more likely to be prefilled or selectable, depending on the password manager and browser.
```
[the remark this answers is not in the document]
Comment (garrett) [reply _posts/2022-09-22-cockpit-277.md:19]: **username**
yoo͞′zər-nām″
_noun_

1. A sequence of characters, different from a password, that is used as identification and is required when logging on to a multiuser computer system, LAN, bulletin board system, or online service.
2. A person's identification on an individual computer system.

_— The American Heritage® Dictionary of the English Language, 5th Edition._

File: _posts/2022-09-22-cockpit-277.md (hunk 1 above)
Comment (garrett) [reply _posts/2022-09-22-cockpit-277.md:29]: Thanks! I think we should specify it in the notes.

People who are having this problem would like to know if they're affected, I think.

File: _posts/2022-09-22-cockpit-277.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+_Thanks to [Jacek Tomasiak](https://github.com/skazi0) for pointing this out and providing a fix!_
+
+## login: Adjust selectors when checking for `:is()` and `:where()` support
```
[the remark this answers is not in the document]
Comment (garrett) [reply _posts/2022-09-22-cockpit-277.md:23]: Sure, that makes sense. We go into detail in the text below.

```suggestion
## Login: Fix "This web browser is too old" error with upcoming browsers
```

File: _posts/2022-09-22-cockpit-277.md (hunk 1 above)
Comment (garrett) [reply _posts/2022-09-22-cockpit-277.md:29]:
```suggestion
Additionally, hotfixes for older versions of Cockpit have been published for various distributions. If you have an error while trying to log in with a new browser on an older Cockpit version, please upgrade your version of Cockpit.

Updates have been issued for the following distributions and versions:

  - ✅ **CentOS Stream 8 & 9**: Fixed.
  - ✅ **Fedora**: Fixed.
  - ⌛ **RHEL 8.6 & 9.0** are affected, fixes are not yet published.
  - ✅ RHEL versions older than 8.6 are *not* affected.
  - ✅ **Debian**: Fixed.
  - ✅ **Ubuntu Kinetic** (current devel series) and the backport for 22.04 LTS got fixed. 
  - ⌛ **Ubuntu 22.04 LTS** is still affected, but will be fixed soon.
```

File: _posts/2022-09-22-cockpit-277.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+The check has now been adjusted so current and upcoming browser versions all pass.
+
+Additionally, hotfixes for older versions of Cockpit have been published for various distributions. If you have an error while trying to log in with a new browser on an older Cockpit version, please upgrade your version of Cockpit.
+
+Updates have been issued for the following distributions and versions:
+
+  - ✅ **CentOS Stream 8 & 9**: Fixed.
+  - ✅ **Fedora**: Fixed in ≥ 36.
+  - ⌛ **RHEL 8.6 & 9.0** are affected, fixes are not yet published.
+  - ✅ RHEL versions older than 8.6 are *not* affected.
```
Comment (garrett) [inline _posts/2022-09-22-cockpit-277.md:36]:
```suggestion
  - ✅ **RHEL** versions older than 8.6 are *not* affected.
```

File: _posts/2022-09-22-cockpit-277.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
+---
+title: Cockpit 277
+author: garrett
+date: '2022-09-22'
+tags: cockpit, machines, podman
+slug: cockpit-277
+category: release
+summary: ''
```
Comment (garrett) [inline _posts/2022-09-22-cockpit-277.md:8]:
```suggestion
summary: 'Login fixes, RHEL downloading improvements, Podman containers visible by default'
```

File: _posts/2022-09-22-cockpit-277.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+Newer, upcoming browser versions have improved next-level support for `:is()` and `:where()` selectors. Cockpit was checking for support an empty usage which passed on browsers in the earlier (current as of writing) implementation. However, browsers have recently updated their parsing support for "[Forgiving Selector Parsing](https://developer.mozilla.org/en-US/docs/Web/CSS/:is#forgiving_selector_parsing)", which caused the newer development versions of Firefox, Chrome, and WebKit to fail this check, preventing the browsers from logging into Cockpit.
+
+The check has now been adjusted so current and upcoming browser versions all pass.
+
+Additionally, hotfixes for older versions of Cockpit have been published for various distributions. If you have an error while trying to log in with a new browser on an older Cockpit version, please upgrade your version of Cockpit.
+
+Updates have been issued for the following distributions and versions:
+
+  - ✅ **CentOS Stream 8 & 9**: Fixed.
+  - ✅ **Fedora**: Fixed.
```
[the remark this answers is not in the document]
Comment (martinpitt) [reply _posts/2022-09-22-cockpit-277.md]: The version in Fedora 35 is still affected. Older Fedoras are EOL.

**Comment by garrett** [comment]
(I think it's ready to merge.)

File: _posts/2022-09-22-cockpit-277.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+summary: 'Login fixes, RHEL downloading improvements, Podman containers visible by default'
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
+We release regularly.
+
+Here are the release notes from Cockpit 277, cockpit-machines 275, and cockpit-podman 54:
+
+
+## login: Improve password manager compatibility
```
Comment (martinpitt) [inline _posts/2022-09-22-cockpit-277.md:17]:
```suggestion
## Login: Improve password manager compatibility
```

Let's make the capitalization consistent.

**Review by martinpitt — APPROVED** [review]
Thanks to all of you! I'll also tweet this out.

**Pull request merged**